### PR TITLE
Run Scraper as Github Action & Improve CI Style Check

### PR DESCRIPTION
**Pull request by diogogmatos**
Created and tested a new workflow in `main.yml` that makes it so that the Calendarium Scraper is executed every 7 days, automatically creating an automated PR with the new and updated shifts. If no differences are detected between versions the pull request is automatically closed.

Also improved the CI Style check so that it automatically formats the code with Prettier, instead of simply saying it is not complying with the Prettier standard.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+name: Calendarium Scraper
+
+on:
+  schedule:
+    # run every 7 days
+    - cron: "0 0 */7 * *"
```
Comment (ruilopesm) [inline .github/workflows/main.yml:6]:
```suggestion
    - cron: "0 0 * * 0"
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v4
+
+      - name: Setup Firefox
+        uses: browser-actions/setup-firefox@v1
+
+      - name: Firefox Version
+        run: firefox --version
```
Comment (ruilopesm) [inline .github/workflows/main.yml:20]: Why is this needed?

Comment (diogogmatos) [reply .github/workflows/main.yml:20]: It's only there for debugging purposes to inform which version of firefox was installed in the process. However, it is naturally not needed for the script to work.

File: .github/workflows/style.yml [hunk 3]
```diff
@@ -19,7 +19,7 @@ jobs:
       - name: Setup node
         uses: actions/setup-node@v3
         with:
-          node-version: ${{ matrix.node }}
+          node-version: 18
```
Comment (ruilopesm) [inline .github/workflows/style.yml:22]: This is already specified by the matrix on line 13.

File: .github/workflows/style.yml [hunk 4]
```diff
@@ -29,10 +29,10 @@ jobs:
             ${{ runner.os }}-node-
 
       - name: Install dependencies
-        run: npm ci
+        run: npm install
```
Comment (ruilopesm) [inline .github/workflows/style.yml:32]: > This command is similar to [npm install](https://docs.npmjs.com/cli/v10/commands/npm-install), except it's meant to be used in automated environments such as test platforms, continuous integration, and deployment -- or any situation where you want to make sure you're doing a clean install of your dependencies.

Source: https://docs.npmjs.com/cli/v10/commands/npm-ci

File: scraper/subjects_short_names.json [hunk 5]
```diff
@@ -0,0 +1,378 @@
+{
```
Comment (ruilopesm) [inline scraper/subjects_short_names.json:1]: This one too.

File: scraper/subjects.json [hunk 6]
```diff
@@ -0,0 +1,754 @@
+[
```
[the remark this answers is not in the document]
Comment (diogogmatos) [reply scraper/subjects.json:1]: We need those so that the subjects that we manually changed in order to prevent situations where the subjects short names are equal. Otherwise the scraper will re-generate those files every time it runs, which both makes it a lot slower and generates filters with problems like duplicated names.

File: scraper/subjects_short_names.json (hunk 5 above)
Comment (diogogmatos) [reply scraper/subjects_short_names.json:1]: Addressed previously.

File: .github/workflows/style.yml (hunk 3 above)
Comment (diogogmatos) [reply .github/workflows/style.yml:22]: Done

File: .github/workflows/style.yml (hunk 4 above)
Comment (diogogmatos) [reply .github/workflows/style.yml:32]: Got it

File: .github/workflows/style.yml (hunk 3 above)
Comment (diogogmatos) [reply .github/workflows/style.yml:22]: True

**Review by ruilopesm — APPROVED** [review]
